pkg/webhook/controlplane/kubeletconfigcodec.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"bytes"
+
+	"github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig/oscommon/cloudinit"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+// KubeletConfigCodec contains methods for encoding and decoding *kubeletconfigv1beta1.KubeletConfiguration objects
+// to and from *extensionsv1alpha1.FileContentInline.
+type KubeletConfigCodec interface {
+	// Encode encodes the given *kubeletconfigv1beta1.KubeletConfiguration into a *extensionsv1alpha1.FileContentInline.
+	Encode(*kubeletconfigv1beta1.KubeletConfiguration, *extensionsv1alpha1.FileContentInline) error
+	// Decode decodes a *kubeletconfigv1beta1.KubeletConfiguration from the given *extensionsv1alpha1.FileContentInline.
+	Decode(*extensionsv1alpha1.FileContentInline, *kubeletconfigv1beta1.KubeletConfiguration) error
+}
+
+// NewKubeletConfigCodec creates an returns a new KubeletConfigCodec.
+func NewKubeletConfigCodec() KubeletConfigCodec {
+	// Create scheme
+	scheme := runtime.NewScheme()
```

Comment:
Instantiate this `scheme` only once since it's always the same (e.g. in an `init` function using a `SchemeBuilder`, then you can also ensure using `utilruntime.Must` that the scheme construction worked).

Reply:
Done.